### ISSUE TITLE
A custom webview editor restores on the wrong editor group when split and restarted

### ISSUE THREAD

**Issue by giammi56** [microsoft/vscode#111851]
Version: 1.51.1 (user setup)
Commit: e5a624b788d92b8d34d1392e4c4d9789406efe8f
Date: 2020-11-10T23:34:32.027Z
Electron: 9.3.3
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Windows_NT x64 10.0.19042
ipython 7.19.0 py38hc5df569_0 conda-forge
ipython_genutils 0.2.0 py_1 conda-forge
Type of virtual environment used: conda
Jupyter server running: Local

Steps to Reproduce:

1.split a workspace and move a tab
2.close and reopen VScode
3. the tabs are all shifted to the right split and the left one is empty

Does this issue occur when all extensions are disabled?: Yes

![image](https://user-images.githubusercontent.com/11757284/101094843-54a32080-35bd-11eb-87dc-2e2ad4fd7073.png)

**Comment by bpasero** [comment]
What are the exact steps? And what is the issue?

**Comment by giammi56** [comment]
Steps to Reproduce:

1.right click on a tab, select "split right" and move a tab to the split workspace
2.close and reopen VScode
3. the tabs are all shifted to the right split and the left one is empty

**Comment by bpasero** [comment]
@giammi56 first of all you say "split a workspace and move a tab", I have absolutely no idea what you mean by "split a workspace". without steps that I can reasonably reproduce on my end, there is nothing for me to do here.

**Comment by giammi56** [comment]
@bpasero the steps are there:

1.right click on a tab, select "split right" and move a tab to the new are created.

what's not clear? what can you not reproduce of this simple action?

**Comment by bpasero** [comment]
Following your steps I end up with the same file as it was in the beginning:

![recording](https://user-images.githubusercontent.com/900690/101146491-58659000-361b-11eb-911d-bdbd55bebb10.gif)

**Comment by giammi56** [comment]
Steps to Reproduce:
Ok, now I understand: try this procedure
0. open multiple tabs (I have .ipynb files, but probably is not related to the file type)
1.right click on a tab, select "split right"
2.close and reopen VScode
3. the tabs are all shifted to the right split and the left one is empty

Could you reproduce it?

**Comment by bpasero** [comment]
@giammi56 tried that, still seems to work fine. But now I wonder if it is indeed related to the file type you open. Does it reproduce for you when you try just with text files? Any chance you can create a screen recording (try with https://gifcap.dev/)

**Comment by giammi56** [comment]
recording a gif doesn't work because the process involves closing the app. here screenshots before and after:

![image](https://user-images.githubusercontent.com/11757284/101152285-248e6880-3623-11eb-9ff6-74b9713d0ea1.png)

![image](https://user-images.githubusercontent.com/11757284/101152405-530c4380-3623-11eb-8356-332ce82e6feb.png)

maybe it targets the delayed opening of the .ipynb to the right split!

**Comment by bpasero** [comment]
Yeah looks like it. 

@mjbvz since this looks like `webview` moving to you. Is it possible that `webview` restore only after the workbench has been opened and then move to the wrong editor group?

@giammi56 since this seems to involve extensions, can you update your steps to indicate your setup (e.g. which extension is responsible for allowing you to open `ipynb` files?

**Comment by giammi56** [comment]
@bpasero I don't what to report: the version of jupyter notebook? which data do you need?

**Comment by giammi56** [comment]
Maybe you'll find this behaviour interesting too: do you suggest me to open a new issue or maybe waiting?
 (gif to big to be uploaded here, please follow this link https://gofile.io/d/OELY1F )

**Comment by bpasero** [comment]
@giammi56 I was asking for the name of the extension you installed to be able to open these jupyter notebooks. 

That second video looks like an issue as well that I suggest to report separately. Is the issue that the editors flicker, did I get that right? I am not entirely sure if this maybe is a bug inside the extension or inside VSCode though.

**Comment by giammi56** [comment]
Extension: Jupiter (microsoft) v2020.11.399280825

The video goes as follows:
1. drag and drop a tab into the right split.
2. the very tab is now unusable: after scrolling it will return to the first line and editing is impossible
3. the flickering is the continuous selection of other tabs: it goes back and forward making any action impossible
4. it returns back to normal when the tab is finally closed.

I will report it separately.

**Comment by mjbvz** [comment]
I can't reproduce this and also have not seen other reports of this occurring 

If you are still seeing this issue, please share the exact steps I can follow to try reproducing it
